lib/generate-icon-components.mjs

```diff
@@ -0,0 +1,46 @@
+import { pascalCase } from 'change-case';
+import { existsSync } from "node:fs";
+import { mkdir, readFile, readdir, rm, writeFile } from "node:fs/promises";
+import { basename, dirname, extname, join } from "node:path";
+import { fileURLToPath } from 'node:url';
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const ICON_FOLDER = join(__dirname, '..', 'public', 'icons')
+const COMPONENT_ICON_FOLDER = join(__dirname, '..', 'addon', 'components', 'icons');
+
+const files = await readdir(ICON_FOLDER);
+const icons = files.filter(file => extname(file) === '.svg').map(svg => basename(svg, '.svg'));
+
+await prepareOutputDir();
+
+const promises = icons.map(svg => {
+  return generateComponent(svg);
+});
+await Promise.all(promises);
+
+async function generateComponent(iconName) {
+  const componentName = pascalCase(iconName);
+
+  const iconContent = await readFile(join(ICON_FOLDER, iconName + '.svg'));
+  const iconContentWithAttributes = iconContent.toString().replace('>', ' ...attributes>'); // We assume the first closing element is the svg element
+
+  const component = `// THIS FILE IS GENERATED. ANY CHANGES TO THIS FILE WILL BE LOST.
+import type { TOC } from '@ember/component/template-only';
+
+export interface ${componentName}IconSignature {
```

Comment:
Since the interface is static, this could live separately and be imported into each file, which would simplify the generated file slightly.

Reply:
Yea, but since all the files are generated it feels a bit strange to store a common type file somewhere, from which they would import. I don't really see an added value, other than DRYing up the generated code a tiny bit, but we would edit the generation script if we would want to make changes anyways, so no real benefit there either. I would prefer to keep it like this for simplicity sake.

Reply:
It was just a similar strange feeling with generating something that is static, but I don't see a strong reason to pick one feeling over the other.